static/app/components/charts/optionSelector.tsx

```diff
@@ -1,157 +1,73 @@
-import {Component, createRef, Fragment} from 'react';
+import {Fragment} from 'react';
 import styled from '@emotion/styled';
-import isEqual from 'lodash/isEqual';
 
-import {InlineContainer, SectionHeading} from 'sentry/components/charts/styles';
-import DropdownBubble from 'sentry/components/dropdownBubble';
-import DropdownButton from 'sentry/components/dropdownButton';
-import {DropdownItem} from 'sentry/components/dropdownControl';
-import DropdownMenu from 'sentry/components/dropdownMenu';
 import FeatureBadge from 'sentry/components/featureBadge';
-import Tooltip from 'sentry/components/tooltip';
+import CompactSelect from 'sentry/components/forms/compactSelect';
 import Truncate from 'sentry/components/truncate';
-import overflowEllipsis from 'sentry/styles/overflowEllipsis';
-import space from 'sentry/styles/space';
 import {SelectValue} from 'sentry/types';
 import {defined} from 'sentry/utils';
 
-const defaultProps = {
-  menuWidth: 'auto',
-};
-
-type Props = {
-  onChange: (value: string) => void;
+type BaseProps = React.ComponentProps<typeof CompactSelect> & {
   options: SelectValue<string>[];
-  selected: string;
   title: string;
   featureType?: 'alpha' | 'beta' | 'new';
-} & typeof defaultProps;
-
-type State = {
-  menuContainerWidth?: number;
 };
 
-class OptionSelector extends Component<Props, State> {
-  static defaultProps = defaultProps;
-
-  state: State = {};
-
-  componentDidMount() {
-    this.setMenuContainerWidth();
-  }
-
-  shouldComponentUpdate(nextProps: Props, nextState: State) {
-    return !isEqual(nextProps, this.props) || !isEqual(nextState, this.state);
-  }
-
-  componentDidUpdate(prevProps: Props) {
-    if (prevProps.selected !== this.props.selected) {
-      this.setMenuContainerWidth();
-    }
-  }
+type SingleProps = BaseProps & {
+  onChange: (value: string) => void;
+  selected: string;
+  multiple?: false;
+};
+type MultipleProps = BaseProps & {
+  multiple: true;
+  onChange: (value: string[]) => void;
+  selected: string[];
+};
```

Comment:
consolidating `optionSelector` and `optionCheckboxSelector` into one component, only difference is the presence of the `multiple` prop and the accepted values for `selected` and `onChange`